packages/webcert/src/feature/careProvider/CareProviderModalContent.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { getUnitStatistics as selectUnitStatistics, getUser } from '../../store/user/userSelectors'
+import { setUnit } from '../../store/user/userActions'
+import { CareUnit, ExpandableTableRow, SimpleTable, Unit } from '@frontend/common'
+
+export const CareProviderModalContent: React.FC = () => {
+  const dispatch = useDispatch()
+  const user = useSelector(getUser)
+  const unitStatistics = useSelector(selectUnitStatistics)
+
+  const getUnitStatistics = (amountOnUnit: number, amountOnOtherUnit?: number, careUnit?: CareUnit) => {
+    return (
+      amountOnUnit +
+      (careUnit && careUnit.units.length > 0 && amountOnOtherUnit !== undefined ? ` (total ${amountOnUnit + amountOnOtherUnit})` : '')
+    ).toString()
+  }
+
+  const getRows = (units: Unit[]) => {
+    return units.map((unit) => {
+      const questionsOnUnit = unitStatistics[unit.unitId].questionsOnUnit
+      const draftsOnUnit = unitStatistics[unit.unitId].draftsOnUnit
+
+      return (
+        <tr key={unit.unitId}>
+          <td>
+            <button className="ic-link iu-ml-700 iu-text-left" type="button" id={unit.unitId} onClick={handleChooseUnit}>
+              {unit.unitName}
+            </button>
+          </td>
+          <td>{getUnitStatistics(questionsOnUnit)}</td>
+          <td>{getUnitStatistics(draftsOnUnit)}</td>
+        </tr>
+      )
+    })
+  }
+
+  const handleChooseUnit = (event: React.MouseEvent) => {
+    const unitId = event.currentTarget.id
+
+    dispatch(setUnit(unitId))
+  }
+
+  return (
```

Comment:
This return statement is long and kind of hard to read. Could you maybe extract some code in functions to make it easier to read?